simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/component/StationComponent.java

```diff
@@ -94,4 +96,23 @@ public SetVariableValidationResult validate(SetVariableDatum setVariableDatum) {
         return new SetVariableValidationResult(setVariableDatum, validationResult);
     }
 
+    /**
+     * Generates report data for all variable in the component
+     *
+     * @param onlyMutableVariables if true, returns only those variables that can be set by the operator
+     * @return list of {@link ReportDatum}
+     */
+    public List<ReportDatum> generateReportData(boolean onlyMutableVariables) {
+        List<ReportDatum> reportData = new ArrayList<>();
+        variableAccessors.values().forEach(accessor -> {
```

Comment:
Reading the code I dont understand why do you need to iterate if onlyMutableVariables == false? Cant you just return empty list?

Reply:
We have a set of variables that can be changed by the operator - mutable variables
Set of variables that can't be changed by the operator - immutable variable

We iterate through all of them every time, and bool flag onlyMutableVariables tells us if we should return only mutable variables or all of them

Reply:
i guess you can use a map or something here to make it more clear, i'll see what i can do.